bin/src/cli.rs

```diff
@@ -80,28 +81,30 @@ pub fn get_upgrade_fd<'a>(matches: &ArgMatches<'a>) -> i32 {
     .parse::<i32>().expect("the file descriptor must be a number")
 }
 
-pub fn upgrade_worker<'a>(matches: &ArgMatches<'a>) -> Option<()> {
-  matches.subcommand_matches("worker").and_then(|worker_matches| {
+pub fn upgrade_worker<'a>(matches: &ArgMatches<'a>) -> Result<Option<()>, anyhow::Error> {
```

Comment:
I think, you should change the signature of the function to `Result<(), anyhow::Error>`. To me the `Option` was here to handle the error without return it

Reply:
right, this should be changed to `Result<(), anyhow::Error>`